lib/internal/modules/esm/get_format.js

```diff
@@ -92,6 +92,12 @@ let typelessPackageJsonFilesWarnedAbout;
 function getFileProtocolModuleFormat(url, context = { __proto__: null }, ignoreErrors) {
   const { source } = context;
   const ext = extname(url);
+  const deduceFormat = (fromSource, fromUrl) => {
```

Comment:
We should declare this at the top level. I'd also maybe name it `determineFormat` or something more specific to explain the cases when it's used.

Reply:
We could also name it what the flag is named, `detectModuleFormat`

Reply:
done